app/helpers/ExerciseConfig/Pipeline/Box/Boxes/base/JavaUtilsTrait.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Helpers\ExerciseConfig\Pipeline\Box;
+
+use App\Helpers\ExerciseConfig\Variable;
+
+trait JavaUtilsTrait
+{
+  protected function constructClasspath(?Variable $jarFiles) {
+    if ($jarFiles && !$jarFiles->isEmpty()) {
+      $classpath = ".";
+      foreach ($jarFiles->getValueAsArray() as $jar) {
+        $classpath .= ":" . $jar;
+      }
```

Comment:
Ever heard of join/import function?

Reply:
Yes and I do not like them :-)

Reply:
I do. And I like them even better since they much shorter to write, much less error prone, and much more efficient (since you copy the string with every iteration twice!).